parquet/src/arrow/arrow_reader/mod.rs

```
@@ -677,7 +695,14 @@ impl<T: ChunkReader + 'static> Iterator for ReaderPageIterator<T> {
         let total_rows = rg.num_rows() as usize;
         let reader = self.reader.clone();
 
-        let ret = SerializedPageReader::new(reader, meta, total_rows, page_locations);
+        let file_decryptor = Arc::new(self.metadata.file_decryptor().clone().unwrap());
+
+        let crypto_context = CryptoContext::new(
+            meta.dictionary_page_offset().is_some(), rg_idx as i16, self.column_idx as i16, file_decryptor.clone(), file_decryptor);
```

The previous approach of using `.to_i16()?` was better as that would raise an error on an invalid cast rather than silently truncate. I think we should either do that or store these as usize and check the conversion to i16 in `create_module_aad`